presto-main/src/test/java/com/facebook/presto/operator/BenchmarkUnnestOperator.java

```diff
@@ -0,0 +1,424 @@
+            typesBuilder.add(nestedTypeOne);
+            channelsBuilder.add(1);
+
+            if (!nestedTypeTwo.equals("NONE")) {
+                Type nestedTypeTwo = getType(metadata, this.nestedTypeTwo).get();
+                typesBuilder.add(nestedTypeTwo);
+                channelsBuilder.add(2);
+            }
+
+            List<Type> types = typesBuilder.build();
+            List<Integer> channels = channelsBuilder.build();
+
+            this.pages = createInputPages(positionsPerPage, typesBuilder.build(), inputGenerator,
+                    primitiveNullsRatioNestedOne, rowNullsRatioNestedOne);
+
+            operatorFactory = new UnnestOperatorFactory(
+                    0,
+                    new PlanNodeId("test"),
+                    channels.subList(0, 1),
+                    types.subList(0, 1),
+                    channels.subList(1, channels.size()),
+                    types.subList(1, types.size()),
+                    true);
+        }
+
+        public Optional<Type> getType(Metadata metadata, String typeString)
+        {
+            if (typeString.equals("NONE")) {
+                return Optional.empty();
+            }
+            TypeSignature signature = TypeSignature.parseTypeSignature(typeString);
+            return Optional.of(metadata.getType(signature));
+        }
+
+        @TearDown
+        public void cleanup()
+        {
+            executor.shutdownNow();
+            scheduledExecutor.shutdownNow();
+        }
+
+        public TaskContext createTaskContext()
+        {
+            return TestingTaskContext.createTaskContext(executor, scheduledExecutor, TEST_SESSION, new DataSize(2, GIGABYTE));
+        }
+
+        public OperatorFactory getOperatorFactory()
+        {
+            return operatorFactory;
+        }
+
+        public List<Page> getPages()
+        {
+            return pages;
+        }
+
+        private static List<Page> createInputPages(int positionsPerPage, List<Type> types, InputGenerator inputGenerator, double primitiveNullsRatioUnnest, double rowNullsRatioUnnest)
+        {
+            ImmutableList.Builder<Page> pages = ImmutableList.builder();
+            int pageCount = TOTAL_POSITIONS / positionsPerPage;
+
+            for (int i = 0; i < pageCount; i++) {
+                Block[] blocks = new Block[types.size()];
+                blocks[0] = inputGenerator.produceBlock(types.get(0), positionsPerPage, 0.0, 0.0);
+                blocks[1] = inputGenerator.produceBlock(types.get(1), positionsPerPage, primitiveNullsRatioUnnest, rowNullsRatioUnnest);
+                if (blocks.length == 3) {
+                    blocks[2] = inputGenerator.produceBlock(types.get(2), positionsPerPage, 0.0, 0.0);
+                }
+                pages.add(new Page(blocks));
+            }
+
+            return pages.build();
+        }
+    }
+
+    @Benchmark
+    public List<Page> unnest(BenchmarkContext context)
```

Comment:
Nit: Most benchmarks in this repo would put the benchmark at the beginning. I'd recommend put this function before the BenchmarkContext.